soci/fbs/ztoc.fbs

```diff
@@ -0,0 +1,48 @@
+namespace ztoc;
+
+table Xattr {
+	key : string;
+	value : string;
+}
+
+table FileMetadata {
+	name : string;
+	type : string;
+	uncompressedOffset : long;
+	uncompressedSize : long;
+	spanStart : int;
+	spanEnd : int;
+	firstSpanHasBits : bool;	// Flag for if there is partial uncompressed data that is stored in the previous byte
+	linkname : string;		// Target name of link (valid for TypeLink or TypeSymlink)
+	mode : long;			// Permission and mode bits
+	uid : int;			// User ID of owner
+	gid : int;			// Group ID of owner
+	uname : string;			// User name of owner
+	gname : string;			// Group name of owner
+
+	modTime  : string;		// Modification time (is formatted in RFC 3339 format,
+					// with sub-second precision added if present: "2006-01-02T15:04:05.999999999Z07:00")
+	devmajor : long;		// Major device number (valid for TypeChar or TypeBlock)
+	devminor : long;		// Minor device number (valid for TypeChar or TypeBlock)
+
+	xattrs : [Xattr];
+}
+
+table ZtocInfo {
+	spanDigests : [string];
+}
+
+table Ztoc {
+	version : string;		// The version of the Ztoc in format <major_version>.<minor_version>, e.g. 1.0
+	buildToolIdentifier : string;
+
+	metadata : [FileMetadata];
```

Comment:
The compression info got its own struct. Why not the file metadata?

Reply:
Done.